### DISCUSSION TITLE
Show remaining time (laundry time etc) in circle value of card

### DISCUSSION THREAD

**Discussion by plimmplamm** [matt8707/hass-config#101] [General]
Hi Matt, first of all -- I love your work and dedication. Big respect.

I have been modifying your code and an very happy so far. One thing I would like to change tho is the laundry timer.
I have a LG smart washer / dryer that has a sensor for remaining time ("sensor.washer_time_display").

How can I get this value into the circle for a new laundry_button_card.yaml?
I see you have used a similar approach for the person card, but can't seemt to get it working.

This is the mockup of the card (tap_action: wake up , doubel tap: start remotely in case it is set up).

![washer_card_mockup](https://user-images.githubusercontent.com/67526476/194779931-503bcd38-70c8-4341-b09e-7f0ee16014ae.jpg)

Thanks in advance for your help! Even if it is just putting me into the right direction.

:) 

PS: this is the sensor i mentioned: sensor.washer_time_display
Defined as:

`     washer_time_display:
       friendly_name: "Washer Time Display"
       value_template: >
         {% if is_state('sensor.washer_run_state', '-') %}
         {% elif is_state('sensor.washer_run_state', 'Standby') %}
           -:--
         {% else %}
           {{ state_attr('sensor.washer', 'remain_time') }}
         {% endif %}`

**Comment by matt8707** [comment]
here's a circle from a sensor https://github.com/matt8707/hass-config/blob/7d7bc29ceb1177848e39207fa95540c3981cfaa6/ui-lovelace.yaml#L206-L222

is `{{ state_attr('sensor.washer', 'remain_time') }}` a timestamp?

**Reply by plimmplamm** [reply]
Hi @matt8707 -- thanks for the quick reply.

I belive so. I can pull it into the laundry-timer (in the sidebar) you have provided.
I see two ways to go about showing the time in the card.

1. Either calculate hrs/min etc like you have done it with the template for persons
2. Or show the timer digits (HH:MM) in the card itself (no circle)

I would prefer option 1 tho. Can you help?

![screenshot](https://user-images.githubusercontent.com/67526476/194790905-b8eaeb49-3f88-480c-a5df-fc30e4b9718a.png)

**Reply by matt8707** [reply]
I need the output to help you

**Reply by plimmplamm** [reply]
Awesome.

Well, the sensor.py states the following:

`    ThinQSensorEntityDescription(
        key=ATTR_REMAIN_TIME,
        name="Remaining time",
        icon="mdi:clock-outline",
        value_fn=lambda x: x.remain_time,
        entity_registry_enabled_default=False,
    ),`

Furthermore the device_helper.py:

`class LGEBaseDevice:
    """A wrapper to monitor LGE devices"""
    def __init__(self, api_device):
        """Initialize the device."""
        self._api = api_device

    @staticmethod
    def format_time(hours, minutes):
        """Return a time in format hh:mm:ss based on input hours and minutes."""
        if not minutes:
            return "0:00:00"

        if not hours:
            int_minutes = int(minutes)
            if int_minutes >= 60:
                int_hours = int(int_minutes / 60)
                minutes = str(int_minutes - (int_hours * 60))
                hours = str(int_hours)
            else:
                hours = "0"

        if int(minutes) < 10:
            minutes = f"0{int(minutes)}"
        remain_time = [hours, minutes, "00"]
        return ":".join(remain_time)`

Does this help? If not, how to get the output?

**Reply by plimmplamm** [reply]
I tried to modify copy of the PERSON specific section in circle.yaml but got stuck...

**Reply by matt8707** [reply]
> Well, the sensor.py states the following:

I meant the state of the attribute sensor.washer "remain_time", can you post that?

> I tried to modify copy of the PERSON specific section in circle.yaml but got stuck...

this is a timestamp that can be parsed and manipulated: 2022-10-10T02:15:00.006506+00:00
but it looks like that integration already parses that like 0:00:00? That's why it doesn't work

**Reply by plimmplamm** [reply]
Alright. Thanks for clarifying.

I can see the following:

<img width="1183" alt="sen2" src="https://user-images.githubusercontent.com/67526476/194793536-c3229356-a793-4973-9366-eb449f3b4a43.png">
<img width="1036" alt="sen1" src="https://user-images.githubusercontent.com/67526476/194793538-3d56806e-0c21-4c63-af45-ead5d805d83b.png">

Is this sufficient?

**Comment by matt8707** [comment]
You've posted 3 different sensors now..
sensor.washer
sensor.washer_time_display
sensor.front_load_washer

It won't work because you have no timestamp
use this code https://github.com/matt8707/hass-config/blob/7d7bc29ceb1177848e39207fa95540c3981cfaa6/ui-lovelace.yaml#L206-L222 and put your jinja2 template as a circle_input

**Reply by plimmplamm** [reply]
Sorry for the confusion, that happened when I switched from the general documentation to my actual sensors. My bad.

Regarding using the monitor speaker template: How do I have to modify that? I fiddled around with it but still am stuck. Thanks for your help @matt8707

**Reply by plimmplamm** [reply]
As mentioned above, I would need to fit the value into the circle, right? What about stating minutes?

**Reply by matt8707** [reply]
you already created a template here

> washer_time_display: friendly_name: "Washer Time Display" value_template: > {% if is_state('sensor.washer_run_state', '-') %} {% elif is_state('sensor.washer_run_state', 'Standby') %} -:-- {% else %} {{ state_attr('sensor.washer', 'remain_time') }} {% endif %}

use that sensor

```js
let x = states['sensor.washer_time_display']; 
if (x) return x.state; 
```

**Reply by matt8707** [reply]
that would be the first step, to convert it to minutes you can do that in your template sensor.

I won't spend time on that but you could split the sensor on : and do math on each part to get minutes

**Reply by plimmplamm** [reply]
Thanks amigo. Will let you (and others know how I go)

**Comment by plimmplamm** [comment]
Just in case someone else is looking for a similar feature -- this is how I have solved it with @matt8707's help:

`          - type: custom:button-card
            entity: sensor.front_load_washer
            name: Washer
            tap_action:
              - call-service: smartthinq_sensors.wake_up
            double_tap_action: !include popup/washer_details.yaml
            template:
              - base
              - icon_washingMachine
            state_display: >
                [[[
                  let x = states['sensor.washer_time_display']; 
                  if (x) return x.state; 
                ]]]`

This shows the remaining time as state on the button (see image). See my comments (above) for more details about where to find the sensor.

![washer2](https://user-images.githubusercontent.com/67526476/194990657-64902d11-fb65-4a92-98c4-45e9d55a131f.png)
